samples/unit-tests/series-timeline/datalabels/demo.js

```diff
@@ -168,3 +168,71 @@ QUnit.test('Timeline: General tests.', function (assert) {
         'The first point is null point.'
     );
 });
+
+
+QUnit.test('Timeline: Styled mode #17614', function (assert) {
+    const chart = Highcharts.chart('container', {
+        chart: {
+            width: 600,
+            height: 400,
+            styledMode: true
+        },
+
+        series: [
+            {
+                showInLegend: true,
+                dataLabels: {
+                    allowOverlap: false
+                },
+                type: 'timeline',
+                data: [
+                    {
+                        x: Date.UTC(1951, 5, 22),
+                        name: 'New Date 1',
+                        label: 'Some new label'
+                    }, {
+                        x: Date.UTC(1957, 9, 4),
+                        name: 'New Date 2',
+                        label: 'Some new label'
+                    }, {
+                        x: Date.UTC(1959, 0, 4),
+                        name: 'New Date 3',
+                        label: 'Some new label'
+                    }, {
+                        x: Date.UTC(1961, 3, 12),
+                        name: 'New Date 3',
+                        label: 'Some new label'
+                    }
+                ]
+            }
+        ]
+    });
+
+    const series = chart.series[0];
+
+    series.points.forEach(point => {
+        const colorIndex = point.colorIndex;
+
+        assert.ok(
+            point.connector
+                .hasClass(`highcharts-color-${colorIndex}`),
+            'Connector has class with the same colorIndex as the point'
+        );
+
+        assert.ok(
+            point.dataLabel.element.innerHTML.includes(
+                '<tspan class="highcharts-color-' + colorIndex +
+                '">● </tspan>'),
+            'Datalabel dot has class with the same colorIndex as the point'
+        );
+    });
+
+    setTimeout(() => {
```

Comment:
I had to put this test in setTimeout: 0 because in Karma the stroke value without the setTimeout was 'none' and in utils, value was correct and the test passed.